src/libponyc/pass/serialisers.c

```diff
@@ -0,0 +1,97 @@
+#include "serialisers.h"
+#include "../type/lookup.h"
+#include "../type/subtype.h"
+#include "../type/reify.h"
+#include <string.h>
+#include <stdio.h>
+#include <assert.h>
+
+static bool entity_serialiser(pass_opt_t* opt, ast_t* entity,
+                              const char* serialise_space, const char* serialise, const char* deserialise)
+{
+  ast_t* ast_serialise_space = ast_get(entity, serialise_space, NULL);
+  ast_t* ast_serialise = ast_get(entity, serialise, NULL);
+  ast_t* ast_deserialise = ast_get(entity, deserialise, NULL);
+
+  if(ast_serialise_space == NULL && ast_serialise == NULL &&
+    ast_serialise_space == NULL)
+    return true;
+
+  if(ast_serialise_space == NULL || ast_serialise == NULL ||
+    ast_serialise_space == NULL)
+  {
+    ast_error(opt->check.errors, entity,
+      "If a custom serialisation strategy exists then _serialise_space, "
+      "_serialise and _deserialise must all be provided");
+    return false;
+  }
+
+  if(ast_id(ast_serialise_space) != TK_FUN && ast_id(ast_serialise) != TK_FUN &&
```

Comment:
The goal of this check is unclear to me. In any case the function will return `true`.